src/SWP/Bundle/ContentBundle/Factory/ArticleFactory.php

```diff
@@ -53,11 +59,13 @@ public function __construct(
         FactoryInterface $baseFactory,
         RouteProviderInterface $routeProvider,
         ArticleProviderInterface $articleProvider,
+        RouteToArticleMapper $routeToArticleMapper,
```

Comment:
an interface should be used instead, because you are coupling the code strictly with `RouteToArticleMapper` class